direct/io-bulk-fs/src/main/java/cz/o2/proxima/direct/bulk/BinaryBlobFormat.java

```diff
@@ -237,75 +244,62 @@ private StreamElement fromBytes(byte[] data) throws IOException {
     }
 
     @Override
-    public void close() throws IOException {
+    public void close() {
       if (blobStream != null) {
-        blobStream.close();
+        ExceptionUtils.unchecked(blobStream::close);
         blobStream = null;
       }
     }
-  }
 
-  /**
-   * Create writer from given {@link OutputStream}.
-   *
-   * @param gzip {@code true} if the output be gzipped
-   * @param out the {@link OutputStream}
-   * @return writer
-   * @throws IOException on IO errors
-   */
-  public static Writer writer(boolean gzip, OutputStream out) throws IOException {
-
-    return new Writer(gzip, out);
+    @Override
+    public Path getPath() {
+      return path;
+    }
   }
 
-  public Writer writer(boolean gzip) throws IOException {
-    return new Writer(gzip, new FileOutputStream(path));
-  }
+  private final boolean writeGzip;
 
-  /**
-   * Create reader from given entity and {@link InputStream}.
-   *
-   * @param entity the entity to read attributes for
-   * @param path path to the stream for logging purposes
-   * @param in the {@link InputStream}
-   * @return reader
-   * @throws IOException on IO errors
-   */
-  public static Reader reader(EntityDescriptor entity, String path, InputStream in)
-      throws IOException {
-
-    return new Reader(entity, path, in);
+  BinaryBlobFormat(boolean writeGzip) {
+    this.writeGzip = writeGzip;
   }
 
-  public Reader reader(EntityDescriptor entity) throws IOException {
-    return new Reader(entity, path.getName(), new FileInputStream(path));
+  @Override
+  public BinaryBlobWriter openWriter(Path path, EntityDescriptor entity) throws IOException {
+    return new BinaryBlobWriter(path, writeGzip, path.writer());
   }
 
-  @Getter private final File path;
+  @Override
+  public String fileSuffix() {
+    return writeGzip ? "blob.gz" : "blob";
```

Comment:
what about other compression formats?

Reply:
Nothing for now, adding more compression format is out of scope of this PR. It would mean adding `CompressionCodec` abstraction. We might do it in the future.